diff --git a/node_modules/prismarine-auth/src/TokenManagers/MinecraftJavaTokenManager.js b/node_modules/prismarine-auth/src/TokenManagers/MinecraftJavaTokenManager.js
index 2c1294e..18105bd 100644
--- a/node_modules/prismarine-auth/src/TokenManagers/MinecraftJavaTokenManager.js
+++ b/node_modules/prismarine-auth/src/TokenManagers/MinecraftJavaTokenManager.js
@@ -94,7 +94,7 @@ class MinecraftJavaTokenManager {
   async fetchEntitlements (accessToken) {
     debug(`[mc] fetching entitlements with ${accessToken.slice(0, 16)}`)
     const headers = { ...fetchOptions.headers, Authorization: `Bearer ${accessToken}` }
-    const entitlements = await fetch(Endpoints.MinecraftServicesEntitlement, { headers }).then(checkStatus)
+    const entitlements = await fetch(Endpoints.MinecraftServicesEntitlement + `?requestId=${crypto.randomUUID()}`, { headers }).then(checkStatus)
     debug(`[mc] got entitlement response: ${entitlements}`)
     return entitlements
   }
diff --git a/node_modules/prismarine-auth/src/common/Constants.js b/node_modules/prismarine-auth/src/common/Constants.js
index b12bcde..d309d64 100644
--- a/node_modules/prismarine-auth/src/common/Constants.js
+++ b/node_modules/prismarine-auth/src/common/Constants.js
@@ -12,7 +12,7 @@ module.exports = {
     XstsAuthorize: 'https://xsts.auth.xboxlive.com/xsts/authorize',
     MinecraftServicesLogWithXbox: 'https://api.minecraftservices.com/authentication/login_with_xbox',
     MinecraftServicesCertificate: 'https://api.minecraftservices.com/player/certificates',
-    MinecraftServicesEntitlement: 'https://api.minecraftservices.com/entitlements/mcstore',
+    MinecraftServicesEntitlement: 'https://api.minecraftservices.com/entitlements/license',
     MinecraftServicesProfile: 'https://api.minecraftservices.com/minecraft/profile',
     MinecraftServicesReport: 'https://api.minecraftservices.com/player/report',
     LiveDeviceCodeRequest: 'https://login.live.com/oauth20_connect.srf',
